src/Workspaces/Core/Portable/Shared/Extensions/ProjectExtensions.cs

```diff
@@ -38,5 +39,14 @@ public static async Task<VersionStamp> GetVersionAsync(this Project project, Can
 
             return version.GetNewerVersion(latestVersion);
         }
+
+        public static string GetLanguageSourceFileExtension(this Project project)
+            => project.Language switch
+            {
+                LanguageNames.CSharp => ".cs",
+                LanguageNames.VisualBasic => ".vb",
+                LanguageNames.FSharp => ".fs",
+                _ => throw ExceptionUtilities.UnexpectedValue(project.Language)
```

Comment:
what happens with typescript.  make sure they don't crash.

Reply:
We already have this somewhere else -- refactorings like add type that generate a full file will be calling into it. But in any case the use I saw should just be asking for the existing extension anyways, since even this isn't quite correct if a language has multiple extensions.

Reply:
Yea, I removed this in favor of just using the existing extension.